tests/store/test_models.py

```diff
@@ -22,13 +22,8 @@ def test_search_entities_none(self, mock_search):
 
     def test_parse_charm_data(self):
         charm = models._parse_charm_data(charm_data)
-        self.assertEqual(
-            charm["archive_url"],
-            "https://api.jujucharms.com/v5/apache2-26/archive",
-        )
-        self.assertEqual(
-            charm["bugs_url"], "https://bugs.launchpad.net/apache2-charm"
-        )
+        self.assertTrue("archive_url" in charm and charm["archive_url"])
```

Comment:
I'm not sure these tests are sufficient. We went from checking a valid value to simply checking if it exists.

Reply:
I would say that checking the URL is pointless, and actually worse.

For the site to fulfill its function, no *specific* URL structure is required. What is required is that the `"archive_url"` points to an archive of the charm, and that the `"bugs_url"` points to a place where you can see bugs. But it is *not* a requirement (as far as I'm concerned) for the `archive_url` to have the form `{charm_name}_{version}/archive`, or for the `bugs_url` to necessarily be on `bugs.launchpad.net`.

Checking that the `bugs_url` actually points to a proper "bugs" place is kinda impossible - that's a rather subjective assessment that can only be made by a human. Checking that `archive_url` points to an archive of the charm *is* possible, but it's possibly not desirable in this case, as it would break the encapsulation of our tests - they wouldn't be able to run offline etc.

More generally, it is actively harmful for us to be testing specifics that aren't actually impactful for the app to work as expected by its clients. It makes the tests very brittle, such that any changes require changing tests, which slow everyone down. This change is a case in point. The back-end should be able to choose that bugs no longer exist on bugs.launchpad.net, or come from a different API endpoint, or that the archive URL structure has changed. Those are all details that are external to this application, and this application shouldn't be testing them. It should just be testing that the properties it needs exist, so that the app can fulfill its functions successfully.

So I think that checking the URLs themselves is a red-herring, giving us the illusion that we are checking things like that the archive exists, without actually checking it. And so, in this case, I much prefer @anthonydillon's version.

Reply:
It's not checking that the provided urls are correct that I'm concerned about but that urls are set at all. In the event the library gets updated we cannot be sufficiently confident that the upgrade was successful if we do not check that the values are indeed set. I'm just not sure that the existence of a key and some `truthy` value are sufficient. 

I've dismissed my review, feel free to land if you feel this is ok.

Reply:
The `and charm["archive_url"]` bit checks that it's not empty.

We could go further and do:

``` python3
import validators

self.assertTrue(validators.url(charm["archive_url"]))
```

If you think that's necessary, I'm happy to support it.